operational_analysis/toolkits/reanalysis_downloading.py

```diff
@@ -0,0 +1,316 @@
+    var_dict=None,
+    save_pathname=None,
+    save_filename=None,
+    apikey_file=None,
+):
+    """
+    This function downloads ERA5 or MERRA2 reanalysis data for a specific lat/lon using the PlanetOS API, returning a
+    dataframe and, optionally, saving the data as a csv file. Only the indicated variables are downloaded and the start
+    and end datetimes of the data are determined based on the provided combination of st_date, en_date, and num_years
+    arguments as follows:
+        1. If st_date and en_date are both defined, they will be used as the final start and end dates but will be
+           adjusted if they are outside of the start and end dates of the PlanetOS dataset
+        2. If st_date and en_date are both undefined, the final end date will be the end of the last full month in the
+           PlanetOS dataset and the start date will be num_years before the end date
+        3. If only st_date is defined, the final end date will be the lesser of num_years after the start date or the
+           last datetime of the PlanetOS dataset
+        4. If only en_date is defined, the final start date will be the greater of num_years before the end date or the
+           first dateitme of the PlanetOS dataset
+
+    Args:
+        dataset (:obj:`string`): Dataset name ("merra2" or "era5")
+        lat (:obj:`float`): Latitude (degrees)
+        lon (:obj:`float`): Longitude (degrees)
+        st_date (:obj:`pandas.Timestamp` or :obj:`string`, optional): Desired start datetime of reanalysis data time
+            series. Defaults to None.
+        en_date (:obj:`pandas.Timestamp` or :obj:`string`, optional): Desired end datetime of reanalysis data time
+            series. Defaults to None.
+        num_years (int, optional): [description]. Desired number of years of reanalysis data. Only used if either
+            st_date or en_date are undefined. Defaults to 20.
+        var_dict (:obj:`dict`, optional): Dictionary mapping the desired reanalysis variable names from PlanetOS to
+            standard OpenOA variable names. If undefined, default variables will be downloaded (U and V wind speeds,
+            temperature, and surface air pressure). Defaults to None.
+        save_pathname (:obj:`string`, optional): The path where the downloaded reanalysis data will be saved (if
+            defined). Defaults to None.
+        save_filename (:obj:`string`, optional): The file name used to save the downloaded reeanalysis data (if
+            defined). Defaults to None.
+        apikey_file (:obj:`string`, optional): The combined path and file name where the PlanetOS API key is saved. If
+            undefined, will assume there is a file called "APIKEY" saved in the operational_analysis.toolkits
+            directory. Defaults to None.
+
+    Returns:
+        :obj:`pandas.DataFrame`: A dataframe containing time series of the requested reanalysis variables
+    """
+
+    # Import PlanetOS apikey
+    if apikey_file is None:
```

Comment:
The shortened version could be: 
```python
apikey = open(os.path.join(os.path.dirname(os.path.abspath(__file__)), apikey_file)).readlines()[0].strip()
```
or, if you use pathlib (another of my concerted efforts for making path objects easier to work with and read):
```python
from pathlib import Path  # this goes in the imports
apikey = open(Path(__file__).parent.resolve() / apikey_file).readlines()[0].strip()
```

Reply:
I wanted to allow the user to specify a different path to the file too, for example if they have the API key saved somewhere else for other PlanetOS purposes. This is why it only assumes the current path if apikey_file is None and otherwise assumes apikey_file includes the full path. I'm open to other suggestions for how to allow this flexibility, though.

Reply:
I think that makes good sense so the user doesn't have to open the file, or include an API key in their workflow. Let's stick with what you did for both better usability and credential security.